docs/reference/src/main/asciidoc/environments.asciidoc

```diff
@@ -361,6 +361,19 @@ In general, an implicit bean archive does not have to contain a beans.xml descri
 
 NOTE: The bean discovery mode of `annotated` is supported from version 2.2.5.Final. Previous versions processed implicit bean archives in the same way as explicit bean archives.
 
+==== Servlet Container Detection
+
+Weld servlet container integration is delivered as a single artifact, so that it's possible to include this artifact in a war and deploy the application to any of the suported servlet containers.
+This approach has advantages but also drawbacks.
+One of them is the fact that Weld must detect the servlet container properly.
```

Comment:
In order to link it together in a 'smoother' way, how about this?
`One of them is the fact that Weld attempts to detect the servlet container automatically.
While this works most of the time, there are few rare cases, when it might be necessary to ...`